common/flatpak-dir.c

```diff
@@ -1153,6 +1154,59 @@ flatpak_dir_remove_oci_files (FlatpakDir   *self,
   return TRUE;
 }
 
+static gchar *
+flatpak_dir_revokefs_fuse_create_mountpoint (const gchar *ref,
+                                             GError     **error)
+{
+  g_autoptr(GFile) cache_dir = NULL;
+  g_auto(GStrv) parts = NULL;
+  g_autofree gchar *cache_dir_path = NULL;
+  g_autofree gchar *mnt_dir = NULL;
+  g_autofree gchar *mountpoint = NULL;
+
+  cache_dir = flatpak_ensure_system_user_cache_dir_location (error);
+  if (cache_dir == NULL)
+    return NULL;
+
+  parts = flatpak_decompose_ref (ref, NULL);
+  cache_dir_path = g_file_get_path (cache_dir);
+  mnt_dir = g_strdup_printf ("%s-XXXXXX", parts[1]);
+  mountpoint = g_mkdtemp_full (g_build_filename (cache_dir_path, mnt_dir, NULL), 0755);
+
+  return g_steal_pointer (&mountpoint);
+}
+
+static gboolean
+flatpak_dir_revokefs_fuse_umount (OstreeRepo **repo,
+                                  GLnxLockFile *lockfile,
+                                  const gchar *mountpoint,
+                                  GError **error)
+{
+  g_autoptr(GSubprocess) fusermount = NULL;
+  g_autoptr(GSubprocessLauncher) launcher = NULL;
+
+  /* Clear references to child_repo as not to leave any open fds. This is needed for
+   * a clean umount operation.
+   */
+  g_clear_pointer (repo, g_object_unref);
+  glnx_release_lock_file (lockfile);
+
+  launcher = g_subprocess_launcher_new (G_SUBPROCESS_FLAGS_NONE);
+  fusermount = g_subprocess_launcher_spawn (launcher,
+                                            error,
+                                            "fusermount", "-u", mountpoint,
+                                            NULL);
+  if (g_subprocess_wait_check (fusermount, NULL, error))
+    {
+      g_autoptr(GFile) mountpoint_file = g_file_new_for_path (mountpoint);
+
+      flatpak_rm_rf (mountpoint_file, NULL, NULL);
```

Comment:
And what if this fails?

Reply:
Introduced error reporting but we don't fail the operations here (i.e. we still return TRUE if we fail to delete the mountpoint).
```
g_autoptr(GError) tmp_error = NULL;

if (!flatpak_rm_rf (mnt_dir_file, NULL, &tmp_error))
  g_warning ("Unable to remove mountpoint directory %s: %s", mnt_dir, tmp_error->message);
```

Reply:
Similar reference is https://github.com/flatpak/flatpak/blob/master/common/flatpak-dir.c#L8908-L8917

Reply:
Personally I think that case is different. In the case you linked to, the app has already been undeployed, so reporting an error would be incorrect. In the case here, we have not yet deployed the new/updated app, so it is meaningful to report an error.

Reply:
> we have not yet deployed the new/updated app, so it is meaningful to report an error. 

And fail the entire operation? :thinking: 
If we have successfully executed `fusermount -u` -> wait_check(), but fail to delete the mountpoint-directory, I don't think it's a major issue to fail the entire operation and prevent the deployment.

Reply:
This is just a personal preference, and this is not my project, but I am not a big fan of warnings.

* If we expect removing the mountpoint directory to always succeed, then failing the entire operation if it fails will only cause problems if the expectation is wrong (in which case, we presumably have a bug)
* If we expect removing the mountpoint directory to fail, then (command-line) users will frequently see a warning, but they will not be able to do anything about it. It's just noise on the terminal (and in the journal)